src/kernels/resize_bilinear.ts

```diff
@@ -0,0 +1,88 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+import {computeDispatch} from '../webgl2compute_util';
+
+import {WebGL2ComputeProgram} from './webgl2compute_program';
+
+export class ResizeBilinearProgram implements WebGL2ComputeProgram {
+  outputShape: number[];
+  userCode: string;
+  dispatchLayout: {x: number[], y: number[], z: number[]};
+  workGroupSize: [number, number, number] = [64, 1, 1];
+  dispatch: [number, number, number];
+  variableNames = ['x'];
+
+  constructor(
+      inputShape: [number, number, number, number], newHeight: number,
+      newWidth: number, alignCorners: boolean) {
+    this.outputShape = [inputShape[0], newHeight, newWidth, inputShape[3]];
+
+    this.dispatchLayout = {x: [1], y: [2], z: [0, 3]};
+
+    this.dispatch = computeDispatch(this.dispatchLayout, this.outputShape);
+
+    const adjustHeight = alignCorners && newHeight > 1;
+    const adjustWidth = alignCorners && newWidth > 1;
+
+    this.userCode = `
+      void main() {
+        ivec4 coords = getOutputCoords();
+
+        if (all(lessThan(coords, outShape))) {
+          int b = coords[0];
+          int d = coords[3];
+          ivec2 rc = coords.yz;
+
+          vec2 effectiveInSize = vec2(
+            ${adjustHeight ? 'xShape.y - 1' : 'xShape.y'},
+            ${adjustWidth ? 'xShape.z - 1' : 'xShape.z'});
+
+          vec2 effectiveOutSize = vec2(
+            ${adjustHeight ? 'outShape.y - 1' : 'outShape.y'},
+            ${adjustWidth ? 'outShape.z - 1' : 'outShape.z'});
+
+          vec2 effectiveInputOverOutputRatioRC =
+              effectiveInSize / effectiveOutSize;
+
+          // Fractional source index
+          vec2 sourceFracIndexRC = vec2(rc) * effectiveInputOverOutputRatioRC;
+
+          // Compute the four integer indices.
+          ivec2 sourceFloorRC = ivec2(sourceFracIndexRC);         
```

Comment:
I see extra spaces at the end of line. Please remove them.

Reply:
It seems the previous yarn lint can not detect trailing spaces, so I add no-trailing-whitespace to fix this. PTAL

Reply:
Thanks Xing. I think `yarn lint` is very helpful. But please make sure that you have resolved all the `yarn lint` error. I see lots of errors in yarn lint. For example:
```
ERROR: (no-any) D:/workspace/jiajia/tfjs/tfjs-webgl2compute/src/backend_webgl2compute.ts[219, 21]: Type declaration of 'any' loses type-safety. Consider replacing it with a more precise type.
ERROR: (no-any) D:/workspace/jiajia/tfjs/tfjs-webgl2compute/src/backend_webgl2compute.ts[220, 17]: Type declaration of 'any' loses type-safety. Consider replacing it with a more precise type.
ERROR: (no-any) D:/workspace/jiajia/tfjs/tfjs-webgl2compute/src/backend_webgl2compute.ts[220, 49]: Type declaration of 'any' loses type-safety. Consider replacing it with a more precise type.
ERROR: (max-line-length) D:/workspace/jiajia/tfjs/tfjs-webgl2compute/src/kernels/conv2d_mm.ts[83, 1]: Exceeds maximum line length of 80
ERROR: (max-line-length) D:/workspace/jiajia/tfjs/tfjs-webgl2compute/src/kernels/conv2d_naive.ts[60, 1]: Exceeds maximum line length of 80
ERROR: (no-any) D:/workspace/jiajia/tfjs/tfjs-webgl2compute/src/kernels/webgl2compute_program.ts[83, 41]: Type declaration of 'any' loses type-safety. Consider replacing it with a more precise type.

error Command failed with exit code 2.

Reply:
I didn't edit these files this time. Let's try to fix it in next CL.

Reply:
It's a little weird for me that you add 'yarn lint' support with many lint errors. It's ok to  split this patch into two. One is or bilinear support. The other is for yarn lint support and with all lint error resolved. How do you think?

Reply:
done